qcodes/instrument_drivers/Keysight/Infiniium.py

```diff
@@ -225,11 +267,27 @@ def __init__(self, name, address, timeout=20, **kwargs):
 
         super().__init__(name, address, timeout=timeout,
                          terminator='\n', **kwargs)
+
+        model = self.IDN()['model']
+        max_acquisition_rate = {
+            'MSO8104A': 4e9,
+            'MSOS104A': 20e9,
+        }
+        allowed_acquisition_rates = {
+            'MSO8104A': vals.Enum(*np.sort(
+                np.append([5e7, 1e8, 1.25e8, 2e8, 2.5e8, 5e8,
+                           1e9, 2e9, 4e9],
+                          np.kron([0.5, 1, 2, 2.5, 4],
+                                  10**np.arange(8))))),
+        }
+        self._max_acquisition_rate = max_acquisition_rate.get(model, inf)
+        self._allowed_acquisition_rates = allowed_acquisition_rates.get(
+            model, Numbers(0, self._max_acquisition_rate))
```

Comment:
@WilliamHPNielsen it is a validator though... I confess that it is a slightly confusing choice of name from my side. I just didn't want to make the name any longer... I will call it `self._acquisition_rates_validator` instead to avoid confusion.